packages/nvidia_nat_langchain/tests/langsmith/test_langsmith_integration.py

```
@@ -58,143 +123,223 @@ async def test_eval_callback_creates_dataset_runs_and_feedback(
+        try:
+            ds = langsmith_client.read_dataset(dataset_name=dataset_name)
+        except Exception as e:
+            last_exception = e
+            await asyncio.sleep(0.1)
+            attempts += 1
+
+    assert ds is not None, f"Failed to read dataset {dataset_name}: {last_exception}"
+    cleanup_datasets.append(ds.id)
     examples = list(langsmith_client.list_examples(dataset_id=ds.id))
     assert len(examples) == 2
+    {example.inputs["nat_item_id"] for example in examples} == {"q1", "q2"}
```

Ruff (0.15.15)

[warning] Pointless comparison. Did you mean to assign a value? Otherwise, prepend `assert` or remove it.

(B015)